### PR TITLE
[Improve][Connector-V2]Improve file scan

### PR DESCRIPTION
**Pull request by hellof3**
### Purpose of this pull request
Improving performance during file scanning can significantly reduce the time taken to scan partition table files

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java [hunk 1]
```diff
@@ -89,34 +90,36 @@ boolean checkFileType(String path) {
...
     @Override
     public List<String> getFileNamesByPath(String path) throws IOException {
+        ArrayList<String> fileNames = new ArrayList<>();
+        Path basePath = new Path(path);
+        if (readPartitions.isEmpty()) {
+            fileNames.addAll(getFileNamesByPath(basePath));
+        } else {
+            for (String readPartition : readPartitions) {
+                fileNames.addAll(getFileNamesByPath(new Path(basePath, readPartition)));
+            }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:100]: This way will missed some partition.
eg:
```
readPartition = ["age=11"]
filepath = basePath/location=newyork/age=11
```

Seem like our test case not enough. Could you add some test case to cover this case?

Comment (hellof3) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:100]: I think if you have multi partitions, the partition name should be written in full. There is no 'age=11' partition. We can support wildcard matching, which can be written as 'location=\*/age=11' or '\*/age=11'. What do you think?"

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:100]: I mean, we supported the partition value like `["age=11"]` before, but after this change, it does not supported any more. It's unaccepted.

Comment (hellof3) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java:100]: Ok, I understand

**Comment by EricJoy2048** [comment]
Is there any new progress in this PR?

**Comment by hailin0** [comment]
No update, temporarily close this pr

**Pull request closed without merging**